tsweb/promvarz/promvarz.go

```diff
@@ -14,6 +14,13 @@ import (
 	"tailscale.com/tsweb/varz"
 )
 
+// OmitPromethusMetrics, if set to true, makes Handler not include native
+// Prometheus metrics.
+//
+// This is useful in some specific cases where the built-in Prometheus
+// collectors have poor performance characteristics.
+var OmitPromethusMetrics bool
+
```

Comment:
Note that the whole purpose of `promvarz` is to be a drop-in replacement for `tailscale.com/tsweb/varz`, exporting native Prometheus metric in addition to expvar-converted ones. In cases where we absolutely don't want that, we can just use `varz.Handler` instead of `promvarz.Handler`.

Also, perhaps we can just disable the [process collector](https://github.com/prometheus/client_golang/blob/36b9f46811b73b12b09ca9b6cf65bc04291db555/prometheus/process_collector.go#L68-L102) in the native Prometheus client by default? We export memory usage metrics via expvar, so the only helpful metric we'll lose is `process_cpu_seconds_total`.

```suggestion
func init() {
	prometheus.Unregister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
}
```

Reply:
Okay, for now I'm going to close this and disable the process collector in `trunkd`, since I'm not 100% sure if other people are relying on this in other internal services. It's currently included for all users of `tsweb`, though, via the debug handler:
https://github.com/tailscale/tailscale/blob/8aa5c3534da70118d8878d7053957a10817052ed/tsweb/debug.go#L54